src/services/MetricsService/FilteredMetricsService.ts

```diff
@@ -0,0 +1,193 @@
+        case EventTypeMetric.ExposedClear:
+          if (eventWithContext.exposureStatus.type === ExposureStatusType.Exposed) {
+            return this.publishEvent(EventTypeMetric.ExposedClear, [
+              [
+                'hoursSinceExposureDetectedAt',
+                String(getHoursBetween(getCurrentDate(), new Date(eventWithContext.exposureStatus.exposureDetectedAt))),
+              ],
+            ]);
+          }
+          break;
+      }
+
+      return Promise.resolve();
+    });
+  }
+
+  sendDailyMetrics(systemStatus: SystemStatus, notificationStatus: Status): Promise<void> {
+    return this.serialPromiseQueue.add(() => {
+      return this.publishOnboardedEventIfNecessary(
+        String(notificationStatus),
+        String(systemStatus === SystemStatus.Active),
+      ).then(() => this.metricsService.sendDailyMetrics());
+    });
+  }
+
+  private publishInstalledEventIfNecessary(): Promise<void> {
+    const publishAndMark = (): Promise<void> => {
+      return this.publishEvent(EventTypeMetric.Installed, [], true).then(() =>
+        this.stateStorage.markInstalledEventAsPublished(),
+      );
+    };
+    return this.stateStorage.isInstalledEventPublished().then(isPublished => {
+      if (isPublished === false) {
+        return publishAndMark();
+      }
+    });
+  }
+
+  private publishOnboardedEventIfNecessary(notificationStatus: string, frameworkenabled: string): Promise<void> {
+    const publishAndMark = (): Promise<void> => {
+      return this.publishEvent(EventTypeMetric.Onboarded, [
+        ['pushnotification', notificationStatus],
+        ['frameworkenabled', frameworkenabled],
+      ]).then(() => this.stateStorage.markOnboardedEventShouldNotBePublished());
+    };
+    return this.stateStorage.shouldOnboardedEventBePublished().then(shouldPublish => {
+      if (shouldPublish) {
+        return publishAndMark();
+      }
+    });
+  }
+
+  private publishEnToggleEventIfNecessary(state: boolean, onboardedDate: Date | undefined): Promise<void> {
+    function shouldPublishEnToggle(): boolean {
+      if (state) {
+        if (onboardedDate && getHoursBetween(onboardedDate, new Date()) > 24) {
+          return true;
+        } else {
+          return false;
+        }
+      } else {
+        return true;
+      }
+    }
+
+    if (shouldPublishEnToggle()) {
+      return this.publishEvent(EventTypeMetric.EnToggle, [['state', String(state)]]);
+    } else {
+      return Promise.resolve();
+    }
+  }
+
+  private publishEvent(
+    eventType: EventTypeMetric,
+    metricPayload: [string, string][],
+    forcePushToServer = false,
+  ): Promise<void> {
+    return this.getRegion().then(region => {
+      return this.metricsService.publishMetric(
+        new Metric(new Date().getTime(), eventType, region, metricPayload),
+        forcePushToServer,
+      );
+    });
+  }
+
+  private getRegion(): Promise<string> {
+    return AsyncStorage.getItem(Key.Region).then(region => {
```

Comment:
Would prefer await syntax here vs the then

Reply:
(see comment above)